src/components/Header/HeaderMessages/HeaderMessages.js

```diff
@@ -1,28 +1,64 @@
 import React from 'react';
+import Modal from '../../../components/Modal/Modal';
+import UpgradePlanForm from '../../UpgradePlanForm/UpgradePlanForm';
 
-const HeaderMessages = ({ alert: { type, message, button } }) => {
-  return (
-    <div className={'messages-container ' + type}>
-      <div className="wrapper">
-        <p>{message}</p>
-        {button ? (
-          button.url ? (
-            <a
-              href={button.url}
-              className="button button--light button--tiny"
-              data-testid="linkButton"
-            >
-              {button.text}
-            </a>
-          ) : (
-            <button className="button button--light button--tiny" data-testid="actionButton">
-              {button.text}
-            </button>
-          )
-        ) : null}
-      </div>
-    </div>
-  );
-};
+class HeaderMessages extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      buyModalIsOpen: false,
+    };
+
+    this.handleOpenBuyModal = () => this.setState({ buyModalIsOpen: true });
+    this.handleCloseBuyModal = () => this.setState({ buyModalIsOpen: false });
+  }
+
+  render() {
+    const alert = this.props.alert;
+    const user = this.props.user;
+    //TODO implement max subscribers modal
+    if (alert.button && alert.button.action === 'validateSubscribersPopup') {
+      return <></>;
+    } else {
+      return (
+        <div className={'messages-container ' + alert.type}>
+          <div className="wrapper">
+            <p>{alert.message}</p>
+            {alert.button ? (
+              alert.button.url ? (
```

Comment:
What do you think about writting it in the following way to avoid nesting?

```
alert.button && alert.button.url ? ( . . . )
: alert.button ? ( . . . )
: null
```

Reply:
Based on my previous comment:

> What do you think, in place of catch the actions that we do not know, catch only the actions that we know.
> 
> I mean, maybe today we do not understand validateSubscribersPopup, but in the future maybe there will be more.

What do you think about this:

```
alert.button && alert.button.url ? ( . . . )
: alert.button && alert.button.action === 'expectedAction') ? ( . . . )
: null
```

Reply:
fixed